.github/workflows/ci.yml

```diff
@@ -0,0 +1,76 @@
+name: CI
+
+# Trigger the workflow on push or pull request, but only for the master branch
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+jobs:
+  cabal:
+    name: ${{ matrix.os }} / ghc ${{ matrix.ghc }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macOS-latest
+        cabal: ["latest"]
+        ghc:
+          - "8.4.4"
+          - "8.6.5"
+          - "8.8.4"
+          - "8.10.2"
+
+    steps:
+    - uses: actions/checkout@v2
+      #if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'
+
+    - uses: actions/setup-haskell@v1.1.4
+      id: setup-haskell-cabal
+      name: Setup Haskell
+      with:
+        ghc-version: ${{ matrix.ghc }}
+        cabal-version: ${{ matrix.cabal }}
+
+    - uses: actions/cache@v1
+      name: Cache cabal-store
+      with:
+        path: ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+        key: ${{ runner.os }}-${{ matrix.ghc }}-cabal
+
+    - name: Build
+      run: |
+        cabal update
+        cabal build all --enable-tests --enable-benchmarks --write-ghc-environment-files=always
+    - name: Test
+      run: |
+        cabal test all --enable-tests
+  stack:
+    name: stack / ghc ${{ matrix.ghc }}
```

Comment:
Where does this `matrix.ghc` come from? Is that the same one from under `cabel:`?

Reply:
Ah, good catch.  It appears that there is no `matrix.ghc` for `stack`.

Let me try to fix this.